src/vmm/src/arch/aarch64/regs.rs

```diff
@@ -276,53 +275,6 @@ impl Aarch64RegisterVec {
     }
 }
 
-impl Versionize for Aarch64RegisterVec {
-    fn serialize<W: std::io::Write>(
-        &self,
-        writer: &mut W,
-        version_map: &VersionMap,
-        target_version: u16,
-    ) -> VersionizeResult<()> {
-        self.inner.serialize(writer, version_map, target_version)
-    }
-
-    fn deserialize<R: std::io::Read>(
-        reader: &mut R,
-        version_map: &VersionMap,
-        source_version: u16,
-    ) -> VersionizeResult<Self>
-    where
-        Self: Sized,
-    {
-        let inner = Aarch64RegisterVecInner::deserialize(reader, version_map, source_version)?;
-        let mut total_size: usize = 0;
-        for id in inner.ids.iter() {
-            let reg_size = reg_size(*id);
-            if RegSize::U2048_SIZE < reg_size {
-                return Err(VersionizeError::Deserialize(
-                    "Failed to deserialize aarch64 registers. Registers bigger then 2048 bits are \
-                     not supported"
-                        .to_string(),
-                ));
-            }
-            total_size += reg_size;
-        }
-        if total_size != inner.data.len() {
-            Err(VersionizeError::Deserialize(
-                "Failed to deserialize aarch64 registers. Sum of registers sizes is not equal to \
-                 registers data length"
-                    .to_string(),
-            ))
-        } else {
-            Ok(Self { inner })
-        }
```

Comment:
I had an offline discussion with @ShadowCurse about this. Apparently, the check for <= 2048 bits made sense only for cases where we would have snapshots from a different version than the one we are loading the snapshot for. But I realize now that the check for the total size of the registers might be important (?)

Reply:
Technically this check is to prevent snapshot tampering, but we already have crc check for it, so I think we can remove it. Also snapshots are generally trusted by us.

Reply:
I remember that we needed this check to get Firecracker to gracefully exit instead of exit with panic if an invalid snapshot file is passed. If that's no longer the case, then we can remove it, but if without this, Firecracker can be made to panic, we should keep it

Reply:
Under which condition would this cause Firecracker to panic? Right now, we only snapshot and restore from the same Firecracker version. Under this condition, I don't see how this could fail. Am I missing something?

Reply:
Oh yeah, I think I remember now. It is because of this [`unreachable`](https://github.com/firecracker-microvm/firecracker/blob/bccf2d4ebaa54265584fb23898fdb43cf92bd75c/src/vmm/src/arch/aarch64/regs.rs#L171 ) If snapshot was modified and the id of some register is changed to have values bigger than 8 in the reg size position then conversion will fail. (8 because 2^8 = 256 bytes -> 2048 bit registers - the max size kernel supports)

Reply:
Ok, I will just re-introduce the manual deserialization. It's not that much of a trouble and we will avoid any surprises in the future.

Reply:
Ok, manual implementations are not as straight-forward as I thought. I will look into it again after I address the rest of the comments.

Reply:
Maybe you can leave the wrapper type for now and we create a `good new issue` to remove it?

Reply:
Ok, done! I combined your idea Egor with an idea of Patrick's where I just serialize manually a tuple including the fields and then deserialize that tuple and perform the checks.